libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -509,4 +462,89 @@ export class CartService {
       return false;
     }
   }
+
+  async getNeedsInput(cartId: string): Promise<GetNeedsInputResponse> {
+    const cart = await this.cartManager.fetchCartById(cartId);
+
+    if (cart.state !== CartState.NEEDS_INPUT) {
+      throw new CartInvalidStateForActionError(
+        cartId,
+        cart.state,
+        'getNeedsInput'
+      );
+    }
+
+    if (!cart.stripeSubscriptionId) {
+      throw new CartSubscriptionNotFoundError(cartId);
+    }
+
+    const subscription = await this.subscriptionManager.retrieve(
+      cart.stripeSubscriptionId
+    );
+    if (!subscription) {
+      throw new CartSubscriptionNotFoundError(cartId);
+    }
+
+    const paymentIntent = await this.subscriptionManager.getLatestPaymentIntent(
+      subscription
+    );
+    if (!paymentIntent) {
+      throw new CartError('no payment intent found for cart subscription', {
+        cartId,
+        subscription: subscription.id,
+      });
+    }
+
+    if (paymentIntent.status === 'requires_action') {
+      return {
+        inputType: NeedsInputType.StripeHandleNextAction,
+        data: { clientSecret: paymentIntent.client_secret },
+      } as StripeHandleNextActionResponse;
+    } else {
+      await this.cartManager.setProcessingCart(cartId);
+      return { inputType: NeedsInputType.NotRequired } as NoInputNeededResponse;
+    }
+  }
+
+  async submitNeedsInputAndRedirect(cartId: string, paymentIntentId?: string) {
```

Comment:
Would it make more sense to call this method something like `continueCheckout` and moving it to the `CheckoutService` so that it's more clear what it's responsibilities are?

Reply:
We could absolutely do this, though I'd love to get @julianpoy 's thoughts here as well